api/src/main/java/me/lokka30/treasury/api/economy/transaction/Transaction.java

```diff
@@ -19,6 +21,8 @@
     private final double newBalance;
     private final double transactionAmount;
     @NotNull
+    private final Currency currency;
```

Comment:
What do you think of saving the Currency UUID, not the Currency instance itself?